src/main/java/de/tum/in/www1/artemis/service/TutorialGroupService.java

```diff
@@ -85,6 +98,188 @@ public Set<StudentDTO> registerMultipleStudents(TutorialGroup tutorialGroup, Set
         return notFoundStudentDTOs;
     }
 
+    /**
+     * Import registrations
+     * <p>
+     * Important to note: A registration must contain a title of the tutorial group, but it must not contain a student.
+     *
+     * <ul>
+     *     <li> Only title -> Create Tutorial Group with the given title if it not exists
+     *     <li> Title and student -> Create Tutorial Group with given title if not exists AND register student.
+     *     <ul>
+     *         <li> If student is already registered in a tutorial group of the same course, the student will be deregistered from the old tutorial group.
+     *     </ul>
+     * </ul>
+     *
+     * @param course        The course to import the registrations for.
+     * @param registrations The registrations to import.
+     * @return The set of registrations with information about the success of the import
+     */
+    public Set<TutorialGroupRegistrationImportDTO> importRegistrations(Course course, Set<TutorialGroupRegistrationImportDTO> registrations) {
```

Comment:
I think it's a bit hard to understand what all these local variables actually are without seeing their types.
Especially "result" often refers to an actual Result but has a completely different type in this method (please specify at least this one)

Reply:
Removed the var in 3bcda0b2fd44ac66ced88f11562d728298c8892a and changed the name of result to make more clear what it represents